### PR TITLE
Core/Totems: update select display for shaman totems

### PR DESCRIPTION
**Pull request by ForesterDev**
**Changes proposed:**

-  Based on 8ca6a20e7303e942adfb6d97b9aa94fab7c6b895, but I use TotemSlot instead of SpellId to reduce DB data

**Target branch(es):** 3.3.5

- [x] 3.3.5

**Tests performed:** (Does it build, tested in-game, etc.) Build, tested in game

### CODE REVIEW

**Comment by ForesterDev** [comment]
Can someone review this PR? It's ready for merge

File: src/server/game/Entities/Totem/Totem.cpp [hunk 1]
```diff
@@ -54,19 +55,22 @@ void Totem::Update(uint32 time)
...
-        WorldPacket data(SMSG_TOTEM_CREATED, 1 + 8 + 4 + 4);
-        data << uint8(m_Properties->Slot - 1);
-        data << uint64(GetGUID());
-        data << uint32(duration);
-        data << uint32(GetUInt32Value(UNIT_CREATED_BY_SPELL));
-        GetOwner()->ToPlayer()->SendDirectMessage(&data);
+        uint32 slot = m_Properties->Slot;
+        if (slot >= SUMMON_SLOT_TOTEM_FIRE && slot < MAX_TOTEM_SLOT)
+        {
+            WorldPacket data(SMSG_TOTEM_CREATED, 1 + 8 + 4 + 4);
```
Comment (ccrs) [inline src/server/game/Entities/Totem/Totem.cpp]: the perfect finish for this PR... port the new packet structure 💃

Comment (ccrs) [reply src/server/game/Entities/Totem/Totem.cpp]: also, wouldnt this fit better in `JustAppeared` ?

Comment (ccrs) [reply src/server/game/Entities/Totem/Totem.cpp]: or maybe not... that would imply to move it to AI... any thoughts @Treeston @Shauren ?

Comment (Treeston) [reply src/server/game/Entities/Totem/Totem.cpp]: AI sending packets gives me twitches. Why do you want it to be in AI?

Comment (ccrs) [reply src/server/game/Entities/Totem/Totem.cpp]: so the packet is sent when the totem is actually visible... makes sense?
let me lookup for any packet log to clarify what the order should be

Comment (ccrs) [reply src/server/game/Entities/Totem/Totem.cpp]: umm realizing now that maybe an override to AddToWorld could be better

Comment (ccrs) [reply src/server/game/Entities/Totem/Totem.cpp]: am I using this review to randomly post my thoughts?..... maybe

Comment (ccrs) [reply src/server/game/Entities/Totem/Totem.cpp]: nvm all this, the packet needs some thought, keep going

**Pull request merged**